src/core/geometry/qgscircle.h

```diff
@@ -366,13 +356,221 @@ class CORE_EXPORT QgsCircle : public QgsEllipse
      */
     QDomElement asGml3( QDomDocument &doc, int precision = 17, const QString &ns = "gml", QgsAbstractGeometry::AxisOrder axisOrder = QgsAbstractGeometry::AxisOrder::XY ) const;
 
+    /**
+     * \brief Method used to calculate the number of segments for circle approximation
+     * \ingroup core
+     * \since QGIS 3.44
+     */
+    enum class SegmentCalculationMethod
+    {
+      Standard = 0,   //!< Standard sagitta-based calculation
+      Adaptive,       //!< Adaptive calculation based on radius size
+      AreaError,      //!< Calculation based on area error
+      ConstantDensity //!< Simple calculation with constant segment density
+    };
+
+    /**
+    * Calculates the number of segments needed to approximate a circle.
+    *
+    * \param radius Circle radius. Must be positive; if <= 0, `minSegments` is returned.
+    * \param parameter Maximum tolerance allowed for the deviation between the circle and its approximation,
+    *        except for the ConstantDensity method where it is a constant. If <= 0, a default value of 0.01 is used.
+    * \param minSegments Minimum number of segments to use. If < 3, it is set to 3.
+    * \param method Calculation method to use.
+    * \returns Number of segments needed for the approximation.
+    *
+    * \pre `radius` must be strictly positive; otherwise, the function returns `minSegments`.
+    * \pre `parameter` should be positive; if not, it defaults to 0.01.
+    * \pre `minSegments` should be at least 3; if less, it is clamped to 3.
```

Comment:
Let's get these nicely formatted in the PyQGIS docs too:

```suggestion
    * \note `radius` must be strictly positive; otherwise, the function returns `minSegments`.
    * \note `parameter` should be positive; if not, it defaults to 0.01.
    * \note `minSegments` should be at least 3; if less, it is clamped to 3.
```